tests/util/event_loop.cpp

```diff
@@ -59,10 +53,17 @@ struct EventLoop::Impl {
     std::mutex m_mutex;
     uv_loop_t* m_loop;
     uv_async_t m_perform_work;
-#elif REALM_PLATFORM_APPLE
+#elif REALM_USE_CF
     Impl(util::CFPtr<CFRunLoopRef> loop) : m_loop(std::move(loop)) { }
 
     util::CFPtr<CFRunLoopRef> m_loop;
+#elif REALM_USE_ALOOPER
+    Impl(ALooper* looper);
+    void perform_work();
+
+    std::vector<std::function<void()>> m_pending_work;
+    std::mutex m_mutex;
+    std::shared_ptr<util::EventLoopSignal<std::function<void()>>> m_signal;
```

Comment:
Why does this need shared ownership? It looks to be owned exclusively by this `Impl` type.

Reply:
The implementation of the Android `EventLoopSignal` relies on it being owned by a `std::shared_ptr`, as far as I can tell.

Reply:
Ah, that explains it!